tests/regression_tests/mg_temperature/build_2g.py

```diff
@@ -0,0 +1,291 @@
+    # Initialized library params
+    group_edges = [0.0, 0.625, 20.0e6]
+    groups = openmc.mgxs.EnergyGroups(group_edges=group_edges)
+    mg_cross_sections_file_micro = openmc.MGXSLibrary(groups)
+    mg_cross_sections_file_macro = openmc.MGXSLibrary(groups)
+    # Building a micro mg library
+    micro_cs = create_micro_xs_dict()
+    for name in names:
+        mg_cross_sections_file_micro.add_xsdata(build_openmc_xs_lib(name,
+                                                                    groups,
+                                                                    [t for t in
+                                                                     micro_cs],
+                                                                    micro_cs))
+    # Building a macro mg library
+    macro_xs = create_macro_dict(micro_cs)
+    mg_cross_sections_file_macro.add_xsdata(build_openmc_xs_lib('macro',
+                                                                groups,
+                                                                [t for t in
+                                                                 macro_xs],
+                                                                macro_xs))
+    # Exporting library to hdf5 files
+    mg_cross_sections_file_micro.export_to_hdf5('micro_2g.h5')
+    mg_cross_sections_file_macro.export_to_hdf5('macro_2g.h5')
+    # Returning the macro_xs dict is needed for analytical solution
+    return macro_xs
+
+
+def analytical_solution_2g_therm(xsmin, xsmax=None, wgt=1.0):
+    """ Calculate eigenvalue based on analytical solution for eq Lf = (1/k)Qf
+        in two group for infinity dilution media in assumption of group
+        boundary in thernmal spectra < 1.e+3 Ev
+        Parametres:
+        ----------
+        xsmin : dict
+            - macro cross-sections dictonary with minimum range temperature
+        xsmax : dict
+            - macro cross-sections dictonary with maximum range temperature
+              by default: None not used for standalone temperature
+        wgt : double
+            - weight for interpolation by default 1.0
+        Returns:
+        ---------
+        keff : np.double - analytical eigenvalue of critical eq matrix
+    """
+    if xsmax is None:
+        sa = xsmin['absorption']['macro']
+        ss12 = xsmin['scatter']['macro'][0][1][0]
+        nsf = xsmin['nu-fission']['macro']
+    else:
+        sa = xsmin['absorption']['macro'] * wgt + \
+             xsmax['absorption']['macro'] * (1 - wgt)
+        ss12 = xsmin['scatter']['macro'][0][1][0] * wgt + \
+               xsmax['scatter']['macro'][0][1][0] * (1 - wgt)
+        nsf = xsmin['nu-fission']['macro'] * wgt + \
+              xsmax['nu-fission']['macro'] * (1 - wgt)
+    L = np.array([sa[0] + ss12, 0.0, -ss12, sa[1]]).reshape(2, 2)
+    Q = np.array([nsf[0], nsf[1], 0.0, 0.0]).reshape(2, 2)
+    arr = np.linalg.inv(L).dot(Q)
+    return np.linalg.eigvals(arr)[1]

```

Comment:
Since the eigenvalues produced from np.linalg.eigvals are not necessarily ordered, would it be better to return np.amax(np.linalg....) instead of indexing with [1]?